widgets/src/main/java/org/pentaho/gwt/widgets/client/listbox/CustomListBox.java

```diff
@@ -937,6 +937,11 @@ public void onKeyDown( Widget widget, char c, int i ) {
     if ( (char) event.getKeyCode() == KeyCodes.KEY_UP || (char) event.getKeyCode() == KeyCodes.KEY_DOWN ) {
       event.preventDefault();
     }
+    if( ( char ) event.getKeyCode() == KEY_TAB || ( char ) event.getKeyCode() == KEY_ESCAPE ){
```

Comment:
If it enters the above `if`, it cannot enter the new one. So, why not `else if`?

Also, this is becoming the use case for a `switch`. Maybe it would be best to use one.

And, btw, I don't think the cast to `char` is needed.